| Задача | Файлы |
| --- | --- |
| 10.1.1 Finally или просто код? | [task1.js](task1.js) |

# 10.1.1 Finally или просто код?
Сравните два фрагмента кода.
Первый использует finally для выполнения кода после try..catch:
try { начать работу работать } catch (e) { обработать ошибку } finally { очистить рабочее пространство }
Второй фрагмент просто ставит очистку после try..catch:
try { начать работу работать } catch (e) { обработать ошибку } очистить рабочее пространство
Нам определённо нужна очистка после работы, неважно возникли ошибки или нет.
Есть ли здесь преимущество в использовании finally или оба фрагмента кода одинаковы? Если такое преимущество есть, то дайте пример, когда оно проявляется.

**Решение**
```javascript
// В данном случае finally вызовется в любом случае, даже если выполнение
// будет завершено через return или throw. Таким образом, преимуществом
// использования finally является гарантированное выполнение блока.

try {
  начать работу
  работать
} catch (e) {
  обработать ошибку
} finally {
  очистить рабочее пространство
}

// В этом случае "очистить рабочее пространство" может быть не выполнено.

try {
  начать работу
  работать
} catch (e) {
  обработать ошибку
}

очистить рабочее пространство
```

